algorand/portal_core.py

```diff
@@ -0,0 +1,550 @@
+                        # Make this block expire.. as long as it is
+                        # not being used to sign itself.  We stick the
+                        # expiration 1000 bytes into the account...
+                        #
+                        # 19200 is approx 24 hours assuming a 4.5 seconds per block   (24 * 3600 / 4.5) = 19200
+                        If(Txn.accounts[3] != Txn.accounts[2],
+                           Pop(blob.write(Int(2), Int(1000), Itob(Txn.first_valid() + Int(19200))))),
+                        blob.meta(Int(3), Bytes("guardian"))
+                    ])],
+                    [a.load() == Int(3), Seq([
+                        off.store(off.load() + Int(1)),
+                        MagicAssert(tchain.load() == Bytes("base16", "0008")),
+                        off.store(off.load() + Int(2) + Int(24)),
+                        fee.store(Btoi(Extract(Txn.application_args[1], off.load(), Int(8)))),
+                        App.globalPut(Bytes("MessageFee"), fee.load()),
+                    ])],
+                    [a.load() == Int(4), Seq([
+                        off.store(off.load() + Int(1)),
+                        MagicAssert(tchain.load() == Bytes("base16", "0008")),
+                        off.store(off.load() + Int(26)),
+                        fee.store(Btoi(Extract(Txn.application_args[1], off.load(), Int(8)))),
+                        off.store(off.load() + Int(8)),
+                        dest.store(Extract(Txn.application_args[1], off.load(), Int(32))),
+
+                        InnerTxnBuilder.Begin(),
+                        InnerTxnBuilder.SetFields(
+                              {
+                                  TxnField.type_enum: TxnType.Payment,
+                                  TxnField.receiver: dest.load(),
+                                  TxnField.amount: fee.load(),
+                                  TxnField.fee: Int(0),
+                              }
+                        ),
+                        InnerTxnBuilder.Submit(),
+                    ])]
+               ),
+                Approve()
+            ])
+
+        def init():
+            return Seq([
+                # You better lose yourself in the music, the moment
+                App.globalPut(Bytes("vphash"), Txn.application_args[2]),
+
+                # You own it, you better never let it go
+                MagicAssert(Txn.sender() == Global.creator_address()),
+
+                # You only get one shot, do not miss your chance to blow
+                MagicAssert(App.globalGet(Bytes("booted")) == Int(0)),
+                App.globalPut(Bytes("booted"), Bytes("true")),
+
+                # This opportunity comes once in a lifetime
+                checkForDuplicate(),
+
+                # You can do anything you set your mind to...
+                hdlGovernance(Int(1))
+            ])
+
+        def verifySigs():
```

Comment:
what's the purpose of this function?

Reply:
I think I understand... this is a dummy function that doesn't do anything... except that it has to be signed by the `vaa_verify` logic sig. So later, you check that the sender for this contract call is indeed the `vaa_verify` sig. Is that right?

Reply:
Which function?  The verifySigs function?  all the processing is done in the logicSig so the code executed in the memPool doesn't have to do anything

Reply:
Yes, that is all correct